sycl/doc/design/CompileTimeProperties.md

```diff
@@ -110,33 +110,55 @@ template </* ... */> class
 
 } // namespace sycl::ext::oneapi
 ```
+An entry is added to the `@llvm.global.annotations` list, as described in
+[IR representation via `@llvm.global.annotations`][5], whenever a 
+[LLVM IR global variable][3] which is associated with a type decorated with
+`[[__sycl_detail__::add_ir_annotations_global_variable()]]` attribute is 
+defined. Associations may include, but are not limited to:
```

Comment:
This last sentence is imprecise.  Don't we need to enumerate the exact scenarios where `add_ir_annotations_global_variable` has an effect?  Otherwise, the front-end team won't know what to implement.